webui/react/src/hooks/useAuthCheck.ts

```diff
@@ -58,14 +65,15 @@ const useAuthCheck = (): (() => void) => {
         if (isAuthFailure(e)) {
           authStore.setAuth({ isAuthenticated: false });
           redirectToExternalSignin();
+          return false;
         }
-        return;
       }
       authStore.setAuth({ isAuthenticated: true });
       authStore.setAuthChecked();
     } else {
       authStore.setAuthChecked();
     }
+    return true;
```

Comment:
These return values don't make sense -- in the `false` path, we've already redirected the browser away from the page, so the code is unreachable.

Reply:
Yeah, I thought the same thing. In testing this though, we actually do reach this or at least we have potential to reach. Under the hood this calls `window.location.replace`. It's not very well documented, but from what I can tell `window.location.replace` is actually asynchronous, which means code after the replace call can execute up to the point where the replace actually happens in the browser. This was causing some unintended side effects in this code which is why the check was added.

Reply:
okay -- can we return `authStore.isAuthenticated.get()` instead, then?

Reply:
Yeah, tested that way and it works. Updated.